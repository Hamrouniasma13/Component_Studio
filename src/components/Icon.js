import React from 'react'
import PropTypes from 'prop-types'
import { ThemeContext } from '@emotion/core'
import { upperFirst, omit } from 'lodash'
import * as icons from '../icons/'

/**
 * Theses props may be passed by parents for styling reason, but we don't want them to be put in the HTML
 * because they throw a warning message.
 */
const PROPS_BLACKLIST = ['left', 'right', 'inline', 'big']

/**
 * List of available icons, based on our icons folder
 */
const AVAILABLES = Object.keys(icons).map(icon =>
  icon.replace('Eb', '').toLowerCase()
)

export default function Icon({ name, color, size = 40, ...props }) {
  const theme = React.useContext(ThemeContext)
  const Icon = icons[upperFirst(name)]
  if (!Icon) return null
  return (
    <Icon
      fill={color ? theme.colors[color] : 'currentColor'}
      width={size}
      height={size}
      {...omit(props, PROPS_BLACKLIST)}
    />
  )
}

Icon.propTypes = {
  name: PropTypes.oneOf(AVAILABLES).isRequired,
  color: PropTypes.string,
  size: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
}

Icon.icons = AVAILABLES
