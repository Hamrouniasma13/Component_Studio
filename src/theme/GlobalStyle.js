import React from 'react'
import { Global } from '@emotion/core'

const STYLES = {
  '*, *:after, *:before': {
    boxSizing: 'border-box'
  },

  'body, input, textarea, select, button': {
    fontFamily: 'Source Sans Pro, sans-serif',
    fontSize: '16px', // OK
    fontWeight: 400 // OK
  },

  'h1, h2, h3, h4, h5, h6, a, p, span, div, input, textarea, button, select': {
    textRendering: 'optimizeLegibility',
    WebkitFontSmoothing: 'antialiased',
    MozOsxFontSmoothing: 'grayscale'
  },

  body: {
    color: '#333',
    margin: 0,
    padding: 0
  },

  '*, *:before, *:after': {
    boxSizing: 'border-box'
  },

  a: {
    color: 'inherit',
    textDecoration: 'none',
    outline: 'none'
  },

  img: {
    maxWidth: '100%'
  }
}

export default () => <Global styles={STYLES} />
