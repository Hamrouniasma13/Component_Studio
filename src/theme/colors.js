const colors = {
  // Main colors
  primary: '#005792', // OK
  secondary: '#F0A202', // OK
  // Text colors
  text: '#333333', // OK

  third: '#36A9E1',
  fourth: '#33B4AE',
  fifth: '#C391BD',
  sixth: '#FBF2C0',
  seventh: '#CE6153',
  label: '#999999',
  charts: ['#457289', '#E7690F', '#C391BD', '#33B4AE', '#36A9E1'],
  // status
  error: '#C91432', // OK
  success: '#1CC54E', // OK
  warning: '#FCD385', // OK

  // icon backgrounds
  background1: '#b5d0ee',
  background2: '#fcd385',
  background3: '#f1afc6',
  background4: '#9fd9b4',

  // Chart colors
  chartColor1: '#025275',
  chartColor2: '#00adc6',
  chartColor3: '#56d1c7',

  // Notifications toast
  toastSuccessText: '#007864',
  toastSuccessBackground: '#96ddd2',
  toastWarningText: '#8F7749',
  toastWarningBackground: '#FCD385',
  toastErrorText: '#a8041f',
  toastErrorBackground: '#eeb8c1',

  // Common colors
  white: '#ffffff',
  black: '#000000',

  // (50) shades of grays
  smallLightGray: '#fafafa',
  lighterGray: '#f3f3f3',
  lightGray: '#e5e5e5',
  gray: '#cccccc',
  darkGray: '#989898',
  darkerGray: '#2a2a2a',
}

export default colors
