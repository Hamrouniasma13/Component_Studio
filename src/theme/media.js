export const breakpoints = {
  mobile: 576,
  tablet: 768,
  laptop: 992,
  desktop: 1200
}

export default {
  mobile: `@media only screen and (min-width : ${breakpoints.mobile}px)`,
  tablet: `@media only screen and (min-width : ${breakpoints.tablet}px)`,
  laptop: `@media only screen and (min-width : ${breakpoints.laptop}px)`,
  desktop: `@media only screen and (min-width : ${breakpoints.desktop}px)`
}
