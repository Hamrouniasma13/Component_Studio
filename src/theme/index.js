import { keyframes } from '@emotion/core'
import { lighten } from './utils'
import colors from './colors'
import eases from './eases'
import media, { breakpoints } from './media'

const progress = keyframes({
  '0%': {
    backgroundPosition: '200%'
  },
  '60%, 100%': {
    backgroundPosition: '0%'
  }
})

const baseColor = '#eee'
const animatedColor = lighten(baseColor, 2)

const theme = {
  colors,
  media,
  breakpoints,
  eases,
  skeletonWave: {
    backgroundImage: `linear-gradient(90deg, ${baseColor} 80%, ${animatedColor} 90%, ${baseColor} 100%)`,
    backgroundSize: '800px',
    animation: `${progress} 3s ease-in-out infinite`,
    backgroundColor: baseColor,
    color: 'transparent',
    display: 'block'
  }
}

export default theme
