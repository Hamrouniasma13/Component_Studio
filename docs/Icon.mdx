---
name: Icon
route: /icon
menu: ⚛️ Atoms
---

import { Playground } from 'docz'
import { Icon } from '../src'

# Icon

<Playground>
  <Icon name="cog" />
</Playground>

## Size

Icons are always _40\*40_ square, but you change the size. Either give just a number, a size in px, or in em, or any other valid css unit.

<Playground>
  <Icon name="cog" size={60} />
</Playground>

## Color

By default, icons take the color of their nearest parent. But you can change it with the `color` prop

<Playground>
  <Icon name="cog" color="primary" />
</Playground>

## List of icons

Here are the icons you can use, _alphabetically ordered_ :

<ul>
  {Icon.icons.sort().map(i => {
    // docz special
    if (i === '__filemeta') return null
    return (
      <li key={i} style={{ marginBottom: 10 }}>
        <Icon size={20} name={i} key={i} />
        <span style={{ margin: '0 10px' }}>:</span>
        <span style={{ fontSize: '16px' }}>{i}</span>
      </li>
    )
  })}
</ul>

## Add an icon

To add an icon, provide the svg file in `assets/icons` and run `yarn icons` to transform the icon you just add into a component we can use.
