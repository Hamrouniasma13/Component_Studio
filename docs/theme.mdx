---
name: Theming
route: /theme
menu: ðŸ§° Technical
---

import { Playground } from 'docz'
import { GlobalStyles, ThemeProvider, Button } from '../src'

# Theming

Theming allows you to fully customize the display of a website.
On our side, you'll be able to only change some colors.

## Theme Provider

_For now_, you **have to** wrap all the components inside the `ThemeProvider` we provide

<Playground>
  <ThemeProvider>My themed App</ThemeProvider>
</Playground>

It will put the default theme in place, which is mandatory for our components to render.

## Change colors

You can change the colors by giving your theme to the `ThemeProvider`

<Playground>
  <ThemeProvider
    theme={{
      colors: {
        primary: '#938293',
      },
    }}
  >
    My custom themed App <Button href="/">Custom button</Button>
  </ThemeProvider>
</Playground>

## Colors

Here is the default colors we provide.

```js
  {
    // Main colors
    primary: '#005792',
    secondary: '#F0A202',

    // Text
    text: '#333333',
    label: '#999999',

    // Additionnal
    third: '#36A9E1',
    fourth: '#33B4AE',
    fifth: '#C391BD',
    sixth: '#FBF2C0',
    seventh: '#CE6153',

    // Status
    error: '#C91432',
    success: '#1CC54E',
    warning: '#FCD385',

    // Background
    background1: '#b5d0ee',
    background2: '#fcd385',
    background3: '#f1afc6',
    background4: '#9fd9b4',

    // Chart
    chartColor1: '#025275',
    chartColor2: '#00adc6',
    chartColor3: '#56d1c7',

    // (50) Shades of grays
    white: '#ffffff',
    smallLightGray: '#fafafa',
    lighterGray: '#f3f3f3',
    lightGray: '#e5e5e5',
    gray: '#cccccc',
    darkGray: '#989898',
    black: '#000000'
  }
```

## Global style

You can also use our `GlobalStyles` component, which will reset some margins and paddings, remove some browser style, and adjust the box-sizing property of every element.
It should be placed once in your app.

<Playground>
  <div>
    <GlobalStyles />
    My App
  </div>
</Playground>
