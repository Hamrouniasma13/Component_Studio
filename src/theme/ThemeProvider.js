import React from 'react'
import { default as defaultTheme } from '../theme'
import { ThemeProvider as EmotionThemeProvider } from 'emotion-theming'
import { merge } from 'lodash'

export const ThemeContext = React.createContext(defaultTheme)

// Used in Admin portal, for color selector
export const ThemeContextProvider = ({ children, theme = {} }) => {
  return (
    <ThemeContext.Provider value={merge({}, defaultTheme, theme || {})}>
      {children}
    </ThemeContext.Provider>
  )
}

// Main theme provider, used in all portals and website studio
export default function ThemeProvider({ children, theme }) {
  return (
    <EmotionThemeProvider theme={merge({}, defaultTheme, theme || {})}>
      {children}
    </EmotionThemeProvider>
  )
}
