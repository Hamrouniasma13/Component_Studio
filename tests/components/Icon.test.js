import React from 'react';
import renderer from 'react-test-renderer';

import theme from '../../theme';
import ThemeProvider from '../../components/ThemeProvider';

import Icon from '../../components/Icon';

describe('Component: <Icon />', () => {
  it('renders default color', () => {
    const tree = renderer.create(<Icon size="50" name="youtube" />);

    expect(tree).toMatchSnapshot();
  });

  it('use theme colors', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <Icon size="50" name="youtube" color="primary" />
      </ThemeProvider>
    );
    expect(tree).toMatchSnapshot();
  });
});
