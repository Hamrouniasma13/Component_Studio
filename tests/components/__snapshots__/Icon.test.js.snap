// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Icon /> renders default color 1`] = `
.emotion-0 {
  color: inherit;
}

<div
  className="emotion-0"
  colorindex={undefined}
  fill="currentColor"
  size="50"
  style={
    Object {
      "alignItems": "center",
      "display": "inline-flex",
      "justifyContent": "center",
    }
  }
>
  <svg
    fill="currentColor"
    height="50"
    style={
      Object {
        "display": "inline-block",
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 16 16"
    width="50"
  >
    <path
      d="M16 9.226l-8-6.21-8 6.21v-2.532l8-6.21 8 6.21zM14 9v6h-4v-4h-4v4h-4v-6l6-4.5z"
    />
  </svg>
</div>
`;

exports[`Component: <Icon /> use theme colors 1`] = `
.emotion-0 {
  color: #494df4;
}

<div
  className="emotion-0"
  colorindex="primary"
  fill="currentColor"
  size="50"
  style={
    Object {
      "alignItems": "center",
      "display": "inline-flex",
      "justifyContent": "center",
    }
  }
>
  <svg
    fill="currentColor"
    height="50"
    style={
      Object {
        "display": "inline-block",
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 16 16"
    width="50"
  >
    <path
      d="M16 9.226l-8-6.21-8 6.21v-2.532l8-6.21 8 6.21zM14 9v6h-4v-4h-4v4h-4v-6l6-4.5z"
    />
  </svg>
</div>
`;
